runtime/src/lib.rs

```diff
@@ -73,13 +78,115 @@ const fn state_version() -> u32 {
     }
 }
 
+const fn consensus_denomination_symbol() -> &'static str {
+    if is_devnet() || is_testnet() {
+        "TEST"
+    } else {
+        "ROSE"
+    }
+}
+
 /// Determine the consensus denomination used by the runtime, depending on
 /// whether the build is for Testner or Mainnet.
 fn consensus_denomination() -> Denomination {
-    if is_devnet() || is_testnet() {
-        "TEST".parse().unwrap()
-    } else {
-        "ROSE".parse().unwrap()
+    consensus_denomination_symbol().parse().unwrap()
+}
+
+/// ERC-20 wrapper around the NATIVE token.
+#[derive(Default)]
+pub struct NativeTokenErc20 {}
+
+impl erc20::AccountToken for NativeTokenErc20 {
+    type Accounts = modules::accounts::Module;
+
+    const GAS_COSTS: erc20::TokenOperationCosts = erc20::TokenOperationCosts::default();
+
+    // keccak-256(RLP("pontusx-paratime/precompiles/erc20:EUROe"))[-20:]
+    const ADDRESS: H160 = H160([
+        0xaa, 0x1e, 0x08, 0x97, 0x83, 0x9c, 0x9b, 0xb9, 0x92, 0x84, 0xa0, 0xb9, 0x85, 0xc5, 0xa3,
+        0xfd, 0x3e, 0xf3, 0x1d, 0x0b,
+    ]);
+
+    const NAME: &str = "EUROe";
+    const SYMBOL: &str = "EUROe";
+    const DECIMALS: u8 = 18;
+
+    fn denomination() -> Denomination {
+        Denomination::NATIVE
+    }
+
+    fn is_minting_allowed(caller: &H160, _address: &H160) -> Result<bool, erc20::Error> {
+        Ok(caller == &H160::from_str("0x24D68bFBA0fB06ccFfD21dC3a5c0B65207Bd479a").unwrap())
+    }
+
+    fn is_burning_allowed(caller: &H160, address: &H160) -> Result<bool, erc20::Error> {
+        // Same access control as for minting.
```

Comment:
This is dangerous. If I get a task to change policy for minting, I will definitely forget to fix this one. So either copy past the policy or create a helper function and use it in both places.